### PR TITLE
Add quantity updating and fix complete order tests

### PR DESCRIPTION
**Pull request by electriceeli**
# Description
Added functions to actively update quantity attributes on product table when orders are completed

Updated example data in module and resolved update function on this.changes to more accurately test function. Added testing for new quantity control functions

## Related Ticket(s)
#6 

## Problem to Solve
Fix testing for complete orders testing module
Update quantity for items sent through completed orders

## Expected Behavior
All tests should pass when ran ONCE. 
`npm run db:reset` must be called before retesting 

## Steps to Test Solution - Testing
`npm run db:reset`
`./node_modules/mocha/bin/mocha`

## Steps to Test Solution - Quantity Feature
`npm run db:reset`
`npm i -g`
`bangazon`
`2`
`1`
`5`
`Y`
`[choose from available account numbers]`
Review database for updated products related to order related to customer  id 1.

## Testing
[x] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[x] I certify that all existing tests pass

### CODE REVIEW

File: app/controllers/completeOrderCtrl.js [hunk 1]
```diff
@@ -35,28 +35,44 @@ module.exports.promptCompleteOrder = (total, paymentReg, payTypes, custId) => {
       }
     }
   };
-  return new Promise ((resolve, reject) => {
+  return new Promise((resolve, reject) => {
     prompt.get(selectReady, function (err, result) {
       switch (result.response) {
         case "Y": {
-          for( let i in payTypes) {
+          for (let i in payTypes) {
```
Comment (TimAConner) [inline app/controllers/completeOrderCtrl.js:42]: A great place to use for (let x of....)! :D

File: app/controllers/completeOrderCtrl.js [hunk 2]
```diff
@@ -35,28 +35,44 @@ module.exports.promptCompleteOrder = (total, paymentReg, payTypes, custId) => {
...
+                    const removeDuplicatesPromises = [];
+                    removeDuplicates.forEach(item => {
+                      removeDuplicatesPromises.push(checkProductQuantity(nullOrders[0].order_id, item));
+                    });
+                    Promise.all(removeDuplicatesPromises).then(result => {
+                      result.forEach(result => {
+                        let final = result.inventory - result.cart_quantity;
+                        updateProductQuantity(final, result.product_id)
+                          .then(results => {
+                            resolve(results);
```
Comment (TimAConner) [inline app/controllers/completeOrderCtrl.js]: Do you need to resolve on each loop of the for each?

**Review by TimAConner — DISMISSED** [review]
Code works, but when I ran the tests, a few failed.  I slacked you the tests that failed and the errors.

**Review by iiimosley — APPROVED** [review]
After a long and turbulent journey, I greatly approve of this merge. Godspeed to us all 🚀

**Review by dhazlettjr — APPROVED** [review]
all clear

**Review by hunterphillips — APPROVED** [review]
🥇

**Pull request merged**